### PR TITLE
Add avatars to TLF lists

### PR DESCRIPTION
**Pull request by jakob223**
Issue: KBFS-4171

### CODE REVIEW

**Comment by jakob223** [comment]
@chrisnojima rebased on master, TSed, and used `classnames`.

File: shared/common-adapters/avatar-line.stories.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// @flow
```
Comment (chrisnojima) [inline shared/common-adapters/avatar-line.stories.js]: should be ts now

File: shared/common-adapters/avatar-line.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          />
+        ))
+        .reverse()}
+    </Kb.Box2>
+  )
+}
+
+const getTextSize = size => (size >= 48 ? 'BodySmallBold' : 'BodyTinyBold')
+
+const styleMap = [128, 96, 64, 48, 32, 24, 16, 12].reduce(function(styles, size) {
```
Comment (chrisnojima) [inline shared/common-adapters/avatar-line.tsx]: should pull this array into its own thing like
`const avatarSizes: ReadOnlyArray<AvatarSize> = [128,96...`

File: shared/common-adapters/list-item2.tsx [hunk 3]
```diff
@@ -88,8 +95,14 @@ const ListItem = (props: Props) => (
...
           direction="horizontal"
-          className={props.onlyShowActionOnHover ? 'hidden-no-hover' : null}
-          style={props.type === 'Small' ? styles.actionSmallContainer : styles.actionLargeContainer}
+          className={Styles.classNames({
+            'hidden-no-hover': props.onlyShowActionOnHover !== null,
+            'hidden-no-hover-animate': props.onlyShowActionOnHover === 'animate',
+          })}
+          style={Styles.collapseStyles([
+            props.type === 'Small' ? styles.actionSmallContainer : styles.actionLargeContainer,
+            props.onlyShowActionOnHover ? styles.hiddenNoHoverAction : {},
```
Comment (chrisnojima) [inline shared/common-adapters/list-item2.tsx]: prefer the form
`props.onlyShowActionOnHover && styles.hiddenNoHoverAction`

File: shared/fs/browser/rows/common.tsx [hunk 4]
```diff
@@ -12,6 +12,7 @@ export type StillCommonProps = {
   inDestinationPicker?: boolean
   onOpen: () => void
   routePath: I.List<string>
+    animateActions?: boolean | null
```
Comment (chrisnojima) [inline shared/fs/browser/rows/common.tsx]: prettier off?

File: shared/fs/browser/rows/tlf-container.tsx [hunk 5]
```diff
@@ -19,13 +19,10 @@ const mapStateToProps = (state, {tlfType, name}: OwnProps) => ({
   _username: state.config.username,
 })
 
-const mergeProps = (
-  stateProps,
-  dispatchProps,
-  {tlfType, name, routePath, destinationPickerIndex}: OwnProps
-) => {
+const mergeProps = (stateProps, dispatchProps, {tlfType, name, routePath, destinationPickerIndex}) => {
```
Comment (chrisnojima) [inline shared/fs/browser/rows/tlf-container.tsx]: likely want to keep `:OwnProps`

File: shared/fs/browser/rows/tlf-container.tsx [hunk 6]
```diff
@@ -37,6 +34,8 @@ const mergeProps = (
     name,
     path,
     routePath,
+    // Only include the user if they're the only one
+    usernames: usernames.size === 0 ? I.List([stateProps._username]) : usernames,
```
Comment (chrisnojima) [inline shared/fs/browser/rows/tlf-container.tsx]: prefer `.isEmpty()`

**Comment by chrisnojima** [comment]
you merged master in but not the roots maybe? diff is strange

**Comment by jakob223** [comment]
@chrisnojima I guess I rebased this and the branch we are merging into at slightly different times. Should look right now.

**Comment by jakob223** [comment]
CI fail is storyshots.. will fix those together after merging into the `4172` branch

**Pull request merged**